flink-runtime/src/main/java/org/apache/flink/runtime/minicluster/MiniClusterJobDispatcher.java

```diff
@@ -405,22 +407,27 @@ private void decrementCheckAndCleanup() {
 
 		private volatile Throwable runnerException;
 
-		private volatile JobExecutionResult result;
+		private volatile org.apache.flink.runtime.jobmaster.JobExecutionResult result;
 		
 		BlockingJobSync(JobID jobId, int numJobMastersToWaitFor) {
 			this.jobId = jobId;
 			this.jobMastersToWaitFor = new CountDownLatch(numJobMastersToWaitFor);
 		}
 
 		@Override
-		public void jobFinished(JobExecutionResult jobResult) {
-			this.result = jobResult;
+		public void jobFinished(org.apache.flink.runtime.jobmaster.JobExecutionResult result) {
+			this.result = result;
 			jobMastersToWaitFor.countDown();
 		}
 
 		@Override
-		public void jobFailed(Throwable cause) {
-			jobException = cause;
+		public void jobFailed(org.apache.flink.runtime.jobmaster.JobExecutionResult result) {
+			checkArgument(result.getSerializedThrowable().isPresent());
+
+			jobException = result
```

Comment:
Actually it is not needed to store the exception separately because the JobExecutionResult already contains the exception.

Reply:
Then I guess that we should remove it and store instead the `jobmaster.JobExecutionResult`.